gossip/src/duplicate_shred_handler.rs

```diff
@@ -131,6 +141,24 @@ impl DuplicateShredHandler {
                     shred1.into_payload(),
                     shred2.into_payload(),
                 )?;
+                // feature_epoch could only be 0 in tests and new cluster setup.
+                if self
+                    .bank_forks
+                    .read()
+                    .unwrap()
+                    .root_bank()
+                    .feature_set
```

Comment:
This is holding the lock on `bank_forks` for the entirety of `if` check.
You need to release the lock after `root_bank`. Something like:
```rust
let root_bank = self.bank_forks.read().unwrap().root_bank();
if root_bank.feature_set....
```